Sources/Structs/Regex.swift

```diff
@@ -0,0 +1,232 @@
+//
+//  Regex.swift
+//  HandySwift
+//
+//  Created by Frederick Pietschmann on 19.03.18.
+//  Copyright © 2018 Flinesoft. All rights reserved.
+//
+// Taken from https://github.com/sharplet/Regex, then modified to remove some weight.
+
+// TODO: Document
+
+public struct Regex {
+    // MARK: - Properties
+    internal let regularExpression: NSRegularExpression
+
+    // MARK: - Initializers
+    /// Create a `Regex` based on a pattern string.
+    ///
+    /// If `pattern` is not a valid regular expression, an error is thrown
+    /// describing the failure.
+    ///
+    /// - parameters:
+    ///     - pattern: A pattern string describing the regex.
+    ///     - options: Configure regular expression matching options.
+    ///       For details, see `Regex.Options`.
+    ///
+    /// - throws: A value of `ErrorType` describing the invalid regular expression.
+    public init(_  pattern: String, options: Options = []) throws {
+        regularExpression = try NSRegularExpression(
+            pattern: pattern,
+            options: options.toNSRegularExpressionOptions()
+        )
+    }
+
+    // MARK: - Methods: Matching
+    /// Returns `true` if the regex matches `string`, otherwise returns `false`.
+    ///
+    /// - parameter string: The string to test.
+    ///
+    /// - returns: `true` if the regular expression matches, otherwise `false`.
+    public func matches(_ string: String) -> Bool {
+        return !matches(in: string).isEmpty
```

Comment:
I feel like this is enough reason to also introduce the `firstMatch` method and use it here. This implementation will be very slow (even in average) if the underlying text is a long text. Using `firstMatch` here would improve the average performance a lot for many cases. Please also note that the `firstMatch(in:)` should be public as I feel like it's needed as often as `matches(in:) -> [Match])` is.